File: src/main/java/xyz/nietongxue/serviceql/BookResolver.java
package xyz.nietongxue.serviceql;

import com.coxautodev.graphql.tools.GraphQLResolver;
import org.springframework.stereotype.Component;


public class BookResolver implements GraphQLResolver<Book> /* This class is a resolver for the xyz.nietongxue.serviceql.Book "Data Class" */ {

    private AuthorRepository authorRepository;

    public BookResolver(AuthorRepository authorRepository) {
        this.authorRepository = authorRepository;
    }

    public Author author(Book book) {
        return authorRepository.findById(book.getAuthorId());
    }
}

File: src/main/kotlin/xyz/nietongxue/serviceql/book/BookResolver.kt
package xyz.nietongxue.serviceql.book

import com.coxautodev.graphql.tools.GraphQLResolver
import org.springframework.stereotype.Component

@Component
class BookResolver(private val authorRepository: AuthorRepository) : GraphQLResolver<Book> /* This class is a resolver for the xyz.nietongxue.serviceql.book.Book "Data Class" */ {

    fun author(book: Book): Author {
        return authorRepository.findById(book.authorId)
    }
}